packages/sdks/web-js-sdk/src/sdk/oidc/index.ts

```diff
@@ -0,0 +1,291 @@
+} from 'oidc-client-ts';
+import {
+  OIDC_CLIENT_TS_DESCOPE_CDN_URL,
+  OIDC_CLIENT_TS_JSDELIVR_CDN_URL,
+} from '../../constants';
+import { getIdToken } from '../../enhancers/withPersistTokens/helpers';
+import { CoreSdk, OidcConfig, OidcConfigOptions } from '../../types';
+import { hasOidcParamsInUrl, removeOidcParamFromUrl } from './helpers';
+
+type OidcModule = {
+  OidcClient: typeof OidcClient;
+  WebStorageStateStore: typeof WebStorageStateStore;
+};
+
+type SignInResponseStorage = Pick<
+  SigninResponse,
+  'id_token' | 'session_state' | 'profile'
+>;
+
+let scriptLoadingPromise: Promise<OidcModule>;
+
+const simpleHash = (input: string): string => {
+  let hash = 0;
+
+  for (let i = 0; i < input.length; i++) {
+    const char = input.charCodeAt(i);
+    hash = (hash << 5) - hash + char;
+    hash = hash & hash; // Convert to 32-bit integer
+  }
+
+  return Math.abs(hash).toString(16); // Return hash as a positive hexadecimal string
+};
+
+const loadScriptWithFallback = (
+  urls: string[],
+  getEntry: () => OidcModule,
+): Promise<OidcModule> => {
+  return new Promise((resolve, reject) => {
+    if (!urls.length)
+      return reject(new Error('No URLs provided to loadScriptWithFallback'));
+
+    const entry = getEntry();
+    if (entry) return resolve(entry);
+
+    const url = urls.shift();
+
+    const scriptEle = document.createElement('script');
+    scriptEle.src = url;
+    scriptEle.id = simpleHash(url);
+    scriptEle.onload = () => {
+      const entry = getEntry();
+      if (entry) return resolve(entry);
+      throw new Error('Could not get entry after loading script from URL');
+    };
+    scriptEle.addEventListener('error', () => {
+      loadScriptWithFallback(urls, getEntry);
+      scriptEle.setAttribute('data-error', 'true');
+    });
+    document.body.appendChild(scriptEle);
+  });
+};
+
+const loadOIDCModule = async (): Promise<OidcModule> => {
+  try {
+    return import('oidc-client-ts');
+  } catch (e) {
+    return loadScriptWithFallback(
+      [OIDC_CLIENT_TS_DESCOPE_CDN_URL, OIDC_CLIENT_TS_JSDELIVR_CDN_URL],
+      () => window['oidc'],
+    );
+  }
+};
+
+function oidcSignInResToStorage(
+  signInRes: SigninResponse,
+): SignInResponseStorage {
+  return {
+    id_token: signInRes.id_token,
+    session_state: signInRes.session_state,
+    profile: signInRes.profile,
+  };
+}
+
+const getUserFromStorage = (
+  stateUserKey: string,
+): SignInResponseStorage | null => {
+  const user = window.localStorage.getItem(stateUserKey);
+  return user ? JSON.parse(user) : null;
+};
+
+const getOidcClient = async (
+  sdk: CoreSdk,
+  projectId: string,
+  oidcConfig?: OidcConfigOptions,
+) => {
+  if (!scriptLoadingPromise) {
+    scriptLoadingPromise = loadOIDCModule();
+  }
+  const { OidcClient, WebStorageStateStore } = await scriptLoadingPromise;
+
+  if (!OidcClient) {
+    throw new Error(
+      'oidc-client-ts is not installed. Please install it by running `npm install oidc-client-ts`',
```

Comment:
Is this error correct?
if it's not installed, we are trying to fetch it from remote URL

Reply:
this is after the fallback, (to be honest, actually inherited this code from previous PR, but I think its ok)